### PR TITLE
Code clean up and dependencies updates

### PR DESCRIPTION
**Pull request by darkgl0w**
Hi ^^

- Updates the CI config
- Updates the dependencies
- Added some tests
- Updated the documentation with a disclaimer (MySQL 8.x incompatibility)
- Add scripts to run containers with MariaDB, MySQL 5.5 or MySQL 5.7 (to run tests on your database of choice container)
- Fix the onClose hook behavior

#### Checklist
- [x] run `npm run test`
- [x] tests are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,21 +12,21 @@ function fastifyMysql (fastify, options, next) {
...
+    const client = connectionType !== 'connection' ? db.pool : db.connection
 
     if (name) {
       if (!fastify.mysql) {
         fastify.decorate('mysql', {})
       }
+
       if (fastify.mysql[name]) {
-        next(new Error('fastify.mysql.' + name + 'has already registered'))
-        return
```
Comment (Eomm) [inline index.js]: attention, without the return statement we are calling the `next()` function twice with unpredictable behaviours

File: package.json [hunk 2]
```diff
@@ -5,7 +5,9 @@
   "main": "index.js",
   "scripts": {
     "test": "standard && tap test/*.test.js",
-    "mysql": "docker run -d -p 3306:3306 -e MYSQL_ALLOW_EMPTY_PASSWORD=yes --rm mysql"
+    "mariadb": "docker run -d -p 3306:3306 -e MYSQL_ALLOW_EMPTY_PASSWORD=yes --rm mariadb",
```
Comment (Eomm) [inline package.json]: could we put a version also for mariadb? WDYT?

Comment (darkgl0w) [reply package.json]: It should work on all versions (including the latest) but we can set it to 10.1 tag as it is the version that is shipped with Debian ^^

File: index.js (hunk 1 above)
Comment (darkgl0w) [reply index.js]: Oups that was a typo, it should return the `next` callback ^^
I must check the test as the weird timeout I experienced on the master branch after updating the dependencies is occurring again ^^

Comment (darkgl0w) [reply index.js]: Ok there is definitely something going wrong with the plugin,
when it throws an error there is a weird timeout and it seems that the connection is not properly closed (seems like the onClose hook is not fired).

Will dig much more into this tomorrow.

Comment (Eomm) [reply index.js]: I'll give you a hand later today if you need 💪🏻

Comment (darkgl0w) [reply index.js]: Hello ^^,
Problem solved :d
As I guessed it was just a matter of when the `onClose` hook was registered and triggered.

**Comment by darkgl0w** [comment]
It should be good now I think, and now that the package act as it should I will prepare another PR to take on this feature request https://github.com/fastify/fastify-mysql/issues/13 if it is ok :)

Feel free to ping me you see anything that needs to be changed ^^

**Comment by mcollina** [comment]
Can you rebase? There are conflicts.

**Pull request merged**